### DISCUSSION TITLE
Firmata example?

### DISCUSSION THREAD

**Discussion by onnimikki** [mattjlewis/diozero#166] [General]
Hi,

great project.  Is there a rudimentary Firmata + Arduino example around?  Just looking for something to start exploring how diozero works with a Firmata-enabled Arduino.  I've been using Firmata4j up  until now and was curious to see how diozero does things differently.

thanks,
James

**Comment by mattjlewis** [comment]
I would strongly recommend using the diozero APIs with the Firmata provider as per the instructions here: https://www.diozero.com/concepts/remotecontrol.html#firmata-for-microcontrollers

You could use the Firmata API that is exposed by the Firmata provider directly but I wouldn’t recommend it as it is quite low level and not portable across devices. I had to do that recently for testing 1-Wire device support, example here: https://github.com/mattjlewis/diozero/blob/main/diozero-provider-firmata/src/main/java/com/diozero/internal/provider/firmata/adapter/example/OneWireTest.java
